addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -505,13 +505,15 @@ def _get_l10n_in_edi_ewaybill_line_details(self, line, line_tax_details, sign):
             "qtyUnit": line.product_id.uom_id.l10n_in_code and line.product_id.uom_id.l10n_in_code.split("-")[0] or "OTH",
             "taxableAmount": self._l10n_in_round_value(line.balance * sign),
         }
-        if tax_details_by_code.get("igst_rate") or (line.move_id.l10n_in_state_id.l10n_in_tin != line.company_id.state_id.l10n_in_tin):
-            line_details.update({"igstRate": self._l10n_in_round_value(tax_details_by_code.get("igst_rate", 0.00))})
-        else:
-            line_details.update({
-                "cgstRate": self._l10n_in_round_value(tax_details_by_code.get("cgst_rate", 0.00)),
-                "sgstRate": self._l10n_in_round_value(tax_details_by_code.get("sgst_rate", 0.00)),
-            })
+        gst_types = {'cgst', 'sgst', 'igst'}
+        gst_tax_rates = {
+            f"{gst_type}Rate": self._l10n_in_round_value(tax_details_by_code.get(f"{gst_type}_rate"))
```

Comment:
```suggestion
            f"{gst_type}Rate": self._l10n_in_round_value(tax_details_by_code[f"{gst_type}_rate"])
```

Reply:
We will not get all the values always
https://github.com/odoo/odoo/blob/65bd1e8828b492f7b9fcd3a2d43ba88704fc648f/addons/l10n_in_edi/models/account_edi_format.py#L577-L592